runtime/backend/test/backend_options_test.cpp

```diff
@@ -0,0 +1,95 @@
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <executorch/runtime/backend/options.h>
+#include <executorch/runtime/platform/runtime.h>
+
+#include <gtest/gtest.h>
+
+using namespace ::testing;
+using executorch::runtime::BackendOptions;
+using executorch::runtime::Error;
+
+class BackendOptionsTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    // Since these tests cause ET_LOG to be called, the PAL must be initialized
+    // first.
+    executorch::runtime::runtime_init();
+  }
+  BackendOptions<5> options; // Capacity of 5 for testing limits
+};
+
+// Test basic string functionality
+TEST_F(BackendOptionsTest, HandlesStringOptions) {
+  // Set and retrieve valid string
+  options.set_option("backend_type", "GPU");
+  const char* result = nullptr;
+  EXPECT_EQ(options.get_option("backend_type", result), Error::Ok);
+  EXPECT_STREQ(result, "GPU");
+
+  // Update existing key
+  options.set_option("backend_type", "CPU");
+  EXPECT_EQ(options.get_option("backend_type", result), Error::Ok);
+  EXPECT_STREQ(result, "CPU");
+}
+
+// Test boolean options
+TEST_F(BackendOptionsTest, HandlesBoolOptions) {
+  options.set_option("debug", true);
+  bool debug = false;
+  EXPECT_EQ(options.get_option("debug", debug), Error::Ok);
+  EXPECT_TRUE(debug);
+
+  // Test false value
+  options.set_option("verbose", false);
+  EXPECT_EQ(options.get_option("verbose", debug), Error::Ok);
+  EXPECT_FALSE(debug);
+}
+
+// Test integer options
+TEST_F(BackendOptionsTest, HandlesIntOptions) {
+  options.set_option("num_threads", 256);
+  int num_threads = 0;
+  EXPECT_EQ(options.get_option("num_threads", num_threads), Error::Ok);
+  EXPECT_EQ(num_threads, 256);
+}
+
+// Test error conditions
+TEST_F(BackendOptionsTest, HandlesErrors) {
+  // Non-existent key
+  bool dummy_bool;
+  EXPECT_EQ(options.get_option("missing", dummy_bool), Error::NotFound);
+
+  // Type mismatch
+  options.set_option("threshold", 100);
+  const char* dummy_str = nullptr;
+  EXPECT_EQ(options.get_option("threshold", dummy_str), Error::InvalidArgument);
+
+  // Null value handling
+  options.set_option("nullable", static_cast<const char*>(nullptr));
+  EXPECT_EQ(options.get_option("nullable", dummy_str), Error::Ok);
+  EXPECT_EQ(dummy_str, nullptr);
+}
+
+// Test type-specific keys
+TEST_F(BackendOptionsTest, EnforcesKeyTypes) {
+  // Same key name - later set operations overwrite earlier ones
+  options.set_option("flag", true);
+  options.set_option("flag", 123); // Overwrites the boolean entry
+
+  bool bval;
+  int ival;
+
+  // Boolean get should fail - type was overwritten to INT
+  EXPECT_EQ(options.get_option("flag", bval), Error::InvalidArgument);
+
+  // Integer get should succeed with correct value
+  EXPECT_EQ(options.get_option("flag", ival), Error::Ok);
+  EXPECT_EQ(ival, 123);
+}
```

Comment:
Add tests for mutability of the options now that you have mutable view

Reply:
Added